### PR TITLE
Add a description for SoundService.DefaultListenerLocation

### PR DESCRIPTION
**Pull request by rbxphogen**
## Changes

Adds a description for SoundService.DefaultListenerLocation

## Checks

By submitting your pull request for review, you agree to the following:

- [x] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [x] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [x] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

File: content/en-us/reference/engine/classes/SoundService.yaml [hunk 1]
```diff
@@ -86,8 +86,29 @@ properties:
...
-    description: ''
+    summary: |
+      Determines where (if anywhere) to place an `Class.AudioListener` by default.
+    description: |
+      Determines where to place an `Class.AudioListener` by default.
+      If `Class.SoundService.DefaultListenerLocation` is `None`, then no `Class.AudioListener` will be created by default, but `Class.AudioListeners` can be created separately by scripts.
+      If `Class.SoundService.DefaultListenerLocation` is `Camera`, then
+      - an `Class.AudioListener` will be created and parented-to `Class.Workspace.CurrentCamera`
+      - an `Class.AudioDeviceOutput` will be created and parented-to `Class.SoundService`
+      - a `Class.Wire` will be created and parented-to the previously-created `Class.AudioListener`
```
Comment (matthewzhang-rbx) [inline content/en-us/reference/engine/classes/SoundService.yaml:97]:
```suggestion
      - an `Class.AudioListener` will be created and parented to `Class.Workspace.CurrentCamera`
      - an `Class.AudioDeviceOutput` will be created and parented to `Class.SoundService`
      - a `Class.Wire` will be created and parented to the previously-created `Class.AudioListener`
```

File: content/en-us/reference/engine/classes/SoundService.yaml [hunk 2]
```diff
@@ -86,8 +86,29 @@ properties:
...
+      - `Class.Wire.SourceInstance` will be set to the previously-created `Class.AudioListener`, and `Class.Wire.TargetInstance` will be set to the previously-created `Class.AudioDeviceOutput`
+      So, the world will be heard from the perspective (postition and orientation) of your camera.
+
+      If `Class.SoundService.DefaultListenerLocation` is `Character`, then
+      - an `Class.Attachment` will be created and parented-to your `Class.Players.LocalPlayer`'s `Class.Player.Character`'s `Class.Model.PrimaryPart`
+      - an `Class.AudioListener` will be created and parented-to the `Class.Attachment`
+      - an `Class.AudioDeviceOutput` will be created and parented-to `Class.SoundService`
+      - a `Class.Wire` will be created and parented-to the previously-created `Class.AudioListener`
```
Comment (matthewzhang-rbx) [inline content/en-us/reference/engine/classes/SoundService.yaml:105]:
```suggestion
      - an `Class.Attachment` will be created and parented to your `Class.Players.LocalPlayer`'s `Class.Player.Character`'s `Class.Model.PrimaryPart`
      - an `Class.AudioListener` will be created and parented to the `Class.Attachment`
      - an `Class.AudioDeviceOutput` will be created and parented to `Class.SoundService`
      - a `Class.Wire` will be created and parented to the previously-created `Class.AudioListener`
```

File: content/en-us/reference/engine/classes/SoundService.yaml [hunk 3]
```diff
@@ -86,8 +86,29 @@ properties:
...
+    summary: |
+      Determines where (if anywhere) to place an `Class.AudioListener` by default.
+    description: |
+      Determines where to place an `Class.AudioListener` by default.
+      If `Class.SoundService.DefaultListenerLocation` is `None`, then no `Class.AudioListener` will be created by default, but `Class.AudioListeners` can be created separately by scripts.
+      If `Class.SoundService.DefaultListenerLocation` is `Camera`, then
+      - an `Class.AudioListener` will be created and parented-to `Class.Workspace.CurrentCamera`
+      - an `Class.AudioDeviceOutput` will be created and parented-to `Class.SoundService`
+      - a `Class.Wire` will be created and parented-to the previously-created `Class.AudioListener`
+      - `Class.Wire.SourceInstance` will be set to the previously-created `Class.AudioListener`, and `Class.Wire.TargetInstance` will be set to the previously-created `Class.AudioDeviceOutput`
```
Comment (matthewzhang-rbx) [inline content/en-us/reference/engine/classes/SoundService.yaml:98]: The documentation supports nested lists, right? We could indent this one forward to put it under the previous line about the wire

File: content/en-us/reference/engine/classes/SoundService.yaml [hunk 4]
```diff
@@ -86,8 +86,29 @@ properties:
...
+      So, the world will be heard from the perspective (postition and orientation) of your camera.
+
+      If `Class.SoundService.DefaultListenerLocation` is `Character`, then
+      - an `Class.Attachment` will be created and parented-to your `Class.Players.LocalPlayer`'s `Class.Player.Character`'s `Class.Model.PrimaryPart`
+      - an `Class.AudioListener` will be created and parented-to the `Class.Attachment`
+      - an `Class.AudioDeviceOutput` will be created and parented-to `Class.SoundService`
+      - a `Class.Wire` will be created and parented-to the previously-created `Class.AudioListener`
+      - `Class.Wire.SourceInstance` will be set to the previously-created `Class.AudioListener`, and `Class.Wire.TargetInstance` will be set to the previously-created `Class.AudioDeviceOutput`
+      - the previously-created `Class.Attachment` will be updated once-per-frame to face the same direction as `Class.Workspace.CurrentCamera`
```
Comment (matthewzhang-rbx) [inline content/en-us/reference/engine/classes/SoundService.yaml:107]: We could move this line to be indented under the earlier line about the Attachment

File: content/en-us/reference/engine/classes/SoundService.yaml (hunk 1 above)
Comment (fumminger) [reply content/en-us/reference/engine/classes/SoundService.yaml]: Agree

File: content/en-us/reference/engine/classes/SoundService.yaml (hunk 2 above)
Comment (fumminger) [reply content/en-us/reference/engine/classes/SoundService.yaml]: Agree

File: content/en-us/reference/engine/classes/SoundService.yaml [hunk 5]
```diff
@@ -86,8 +86,29 @@ properties:
...
+      Determines where (if anywhere) to place an `Class.AudioListener` by default.
+    description: |
+      Determines where to place an `Class.AudioListener` by default.
+      If `Class.SoundService.DefaultListenerLocation` is `None`, then no `Class.AudioListener` will be created by default, but `Class.AudioListeners` can be created separately by scripts.
+      If `Class.SoundService.DefaultListenerLocation` is `Camera`, then
+      - an `Class.AudioListener` will be created and parented-to `Class.Workspace.CurrentCamera`
+      - an `Class.AudioDeviceOutput` will be created and parented-to `Class.SoundService`
+      - a `Class.Wire` will be created and parented-to the previously-created `Class.AudioListener`
+      - `Class.Wire.SourceInstance` will be set to the previously-created `Class.AudioListener`, and `Class.Wire.TargetInstance` will be set to the previously-created `Class.AudioDeviceOutput`
+      So, the world will be heard from the perspective (postition and orientation) of your camera.
```
Comment (fumminger) [inline content/en-us/reference/engine/classes/SoundService.yaml:99]: The "so," seems overly casual to me.
I might say something like "The result is".

File: content/en-us/reference/engine/classes/SoundService.yaml [hunk 6]
```diff
@@ -86,8 +86,29 @@ properties:
...
+
+      If `Class.SoundService.DefaultListenerLocation` is `Character`, then
+      - an `Class.Attachment` will be created and parented-to your `Class.Players.LocalPlayer`'s `Class.Player.Character`'s `Class.Model.PrimaryPart`
+      - an `Class.AudioListener` will be created and parented-to the `Class.Attachment`
+      - an `Class.AudioDeviceOutput` will be created and parented-to `Class.SoundService`
+      - a `Class.Wire` will be created and parented-to the previously-created `Class.AudioListener`
+      - `Class.Wire.SourceInstance` will be set to the previously-created `Class.AudioListener`, and `Class.Wire.TargetInstance` will be set to the previously-created `Class.AudioDeviceOutput`
+      - the previously-created `Class.Attachment` will be updated once-per-frame to face the same direction as `Class.Workspace.CurrentCamera`
+      So, the world will be heard from the position of your character, while matching the orientation of your camera.
```
Comment (fumminger) [inline content/en-us/reference/engine/classes/SoundService.yaml:108]: Again "So," seems overly casual, and assumes the reader can make the connection between the actions above and the result described here.

**Review by fumminger — CHANGES_REQUESTED** [review]

File: content/en-us/reference/engine/classes/SoundService.yaml (hunk 5 above)
Comment (rbxphogen) [reply content/en-us/reference/engine/classes/SoundService.yaml:99]: Sure thing; does this work?
```suggestion
      As a result, the world will be heard from the perspective (postition and orientation) of your camera.
```

File: content/en-us/reference/engine/classes/SoundService.yaml (hunk 6 above)
Comment (rbxphogen) [reply content/en-us/reference/engine/classes/SoundService.yaml:108]:
```suggestion
      As a result, the world will be heard from the position of your character, while matching the orientation of your camera.
```

File: content/en-us/reference/engine/classes/SoundService.yaml (hunk 3 above)
Comment (matthewzhang-rbx) [reply content/en-us/reference/engine/classes/SoundService.yaml:98]: Hm I would think you indent the bullet one more time but I don't know how legit that is

**Review by fumminger — APPROVED** [review]

**Comment by IgnisRBX** [comment]
Since this is controlled by a `Enum.ListenerLocation`, it's best to put a very brief description of this property here, then fill out the options on the `Enum.ListenerLocation` page.

**Comment by rbxphogen** [comment]
@IgnisRBX I created a PR for those docs [here](https://github.com/Roblox/creator-docs/pull/883/files)

**Review by sofiaconmorales — APPROVED** [review]

**Pull request closed without merging**